### DISCUSSION TITLE
MikroOrm upgrade to v6 throws error Cannot read properties of undefined (reading 'values')

### DISCUSSION THREAD

**Discussion by DaleRest** [mikro-orm/mikro-orm#5553] [General]
### Describe the bug

upgrade from v4 to v5 and v6 and this issue is present in both v5 and v6.  I have isolated to my imported modules. How can i further isolate to identify what is causing this?

For reference my appmodule
`@Module({
  controllers: [AppController],
  imports: [
    ConfigModule.forRoot({
      envFilePath: ['.env','.env.local','.env.deployment'],
      isGlobal: true,
    }),
    ConsoleModule,
    MikroOrmModule.forRoot(),
    ScheduleModule.forRoot(),
    SentryModule.forRootAsync({
      imports: [SharedModule,ConfigModule],
      inject: [ConfigService],
      useFactory: async (configService: ConfigService) => ({
        enabled: configService.get<boolean>("SENTRY_ENABLED"),
        dsn: configService.get<string>("SENTRY_DSN"),
        debug: configService.get<boolean>("SENTRY_DEBUG"),
        environment: configService.get<string>("NODE_ENV"),
        release: null, // must create a release in sentry.io dashboard
        logLevels: ['debug'] //LogLevel.Error, //based on sentry.io loglevel //
      }),
    }),
    BullModule.forRootAsync({
      imports: [SharedModule, ConfigModule],
      useFactory: async (configService: ConfigService) => ({
        redis: {
          host: configService.get<string>("REDIS_SERVER_HOST"),
          port: configService.get<number>("REDIS_SERVER_PORT"),
          db: configService.get<number>("REDIS_SERVER_DATABASE"),
          password: configService.get<string>("REDIS_SERVER_PASSWORD"),
        },
      }),
      inject: [ConfigService],
    }),
    LoggerModule.forRootAsync({
      inject: [ConfigService],
      useFactory: async (configService: ConfigService) => {
        return {
          pinoHttp: configService.get<any>
            ? {
                level: "debug",
                transport : {
                  target: "pino-pretty",                         //new (after error resolve)refer here https://stackoverflow.com/questions/71649204/pinowarning-prettyprint-is-deprecated-look-at-https-github-com-pinojs-pino-p
                  options:{
                    colorize: true,
                    levelFirst: true,
                    translateTime: "UTC:mm/dd/yyyy, h:MM:ss TT Z",
                  },
                }
              }
            : {},
        };
      },
      imports: [ConfigModule],
      // providers: undefined
    }),
    AwsSdkModule.forRootAsync({
      // TODO get rid off fallback env variables (added for correct work in local env)
      defaultServiceOptions: {
        useFactory: (configService: ConfigService) => ({
          credentials: {
            accessKeyId:
              configService.get<string>("AWS_SDK_ACCESS_KEY") ||
              configService.get<string>("MINIO_ACCESS_KEY"),
            secretAccessKey:
              configService.get<string>("AWS_SDK_SECRET_KEY") ||
              configService.get<string>("MINIO_SECRET_KEY"),
          },
          region: configService.get<string>("S3_STORAGE_REGION") || null,
          endpoint: configService.get<string>("S3_STORAGE_ENDPOINT_PRIVATE") || null,
          signatureVersion: "v4",
          s3ForcePathStyle: true, // false value not compatible with MinIO
        }),
        inject: [ConfigService],
        imports: [SharedModule,ConfigModule],
      },
    }),
    paymentsModule,
    SlackUserModule,
    OrganisationModule,
    DashboardModule,
    NestAsyncEventModule,
    SharedModule,
    UsageStatsModule,
    UserModule,
    MailingModule,
    CategoryModule,
    AuthModule,
    PostModule,
    InboxModule,
    MessagingModule,
    NotificationModule,
    DataModule,
  ],
  providers: [ConfigService]
})
export class AppModule {
  configure(consumer: MiddlewareConsumer) {
    consumer.apply(LogsMiddleware).forRoutes('*');
  }
}
`

And my authmodule:

`@Module({
  imports: [
    JwtModule.registerAsync({
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: (configService: ConfigService) => ({
        secretOrPrivateKey: configService.get<string>("JWT_SECRET_KEY"),
        signOptions: {
          expiresIn: configService.get<number>("JWT_EXPIRATION_TIME"),
        },
      }),
    }),
    MikroOrmModule.forFeature({ entities: [...Object.values(moduleEntities), User] }),
    forwardRef(() => UserModule),
    // ConfigService,
    NestAsyncEventModule,
    PassportModule.register({ defaultStrategy: "jwt" }),
  ],
  providers: [AuthService, BannedByIpService, JwtStrategy, ConfigService],
  controllers: Object.values(authControllers),
  exports: [JwtStrategy, PassportModule, AuthService, BannedByIpService],
})
export class AuthModule {}`

### Reproduction

upgrade from v4 to v5 and v6 and this issue is present in both v5 and v6.  Once upgraded and you try and run the application, the application throws an error for entity discovery as per below:
`[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [MikroORM] [info] MikroORM version: 6.2.5
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [MikroORM] [discovery] ORM entity discovery started, using TsMorphMetadataProvider
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [InstanceLoader] MikroOrmModule dependencies initialized +25ms
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [InstanceLoader] SentryModule dependencies initialized +1ms
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [InstanceLoader] PassportModule dependencies initialized +1ms
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [InstanceLoader] HttpModule dependencies initialized +4ms
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [InstanceLoader] HttpModule dependencies initialized +1ms
[Nest] 30312  - 09/05/2024, 2:07:17 pm     LOG [InstanceLoader] ConfigHostModule dependencies initialized +3ms
[Nest] 30312  - 09/05/2024, 2:07:17 pm   ERROR [ExceptionHandler] Cannot read properties of undefined (reading 'values')
TypeError: Cannot read properties of undefined (reading 'values')`

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6

### Node.js version

node.js version 20

### Operating system

windows, linux

### Validations

- [x] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I need to see a reproduction, logs are not reproduction. Also the first step should be finding out the full stack trace as without it this error is not telling us anything.

Maybe look at this example project (which works just fine) and compare what you do differently.

https://github.com/mikro-orm/nestjs-realworld-example-app

And as usual, consult the upgrading guides in detail.

**Reply by jdmevo123** [reply]
Sorry i thought i had the full stack trace, here it is:
```typescript
[Nest] 28284  - 09/05/2024, 4:08:05 pm     LOG [InstanceLoader] MikroOrmModule dependencies initialized +38ms
[Nest] 28284  - 09/05/2024, 4:08:05 pm     LOG [InstanceLoader] SentryModule dependencies initialized +0ms
[Nest] 28284  - 09/05/2024, 4:08:05 pm     LOG [InstanceLoader] PassportModule dependencies initialized +1ms
[Nest] 28284  - 09/05/2024, 4:08:05 pm     LOG [InstanceLoader] HttpModule dependencies initialized +5ms
[Nest] 28284  - 09/05/2024, 4:08:05 pm     LOG [InstanceLoader] HttpModule dependencies initialized +1ms
[Nest] 28284  - 09/05/2024, 4:08:05 pm     LOG [InstanceLoader] ConfigHostModule dependencies initialized +3ms
[Nest] 28284  - 09/05/2024, 4:08:05 pm   ERROR [ExceptionHandler] Cannot read properties of undefined (reading 'values')
TypeError: Cannot read properties of undefined (reading 'values')
    at api\services\node-app\node_modules\@nestjs-plus\discovery\src\discovery.service.ts:68:50
    at arrayMap (api\services\node-app\node_modules\lodash\lodash.js:653:23)
    at map (api\services\node-app\node_modules\lodash\lodash.js:9622:14)
    at Function.flatMap (api\services\node-app\node_modules\lodash\lodash.js:9325:26)
    at new DiscoveryService (api\services\node-app\node_modules\@nestjs-plus\discovery\src\discovery.service.ts:67:7)
    at Injector.instantiateClass (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:365:19)
    at callback (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:65:45)
    at Injector.resolveConstructorParams (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:144:24)
    at Injector.loadInstance (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:70:13)
    at Injector.loadProvider (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:97:9)
    at Injector.lookupComponentInImports (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:289:17)
    at Injector.lookupComponentInParentModules (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:252:33)
    at Injector.resolveComponentInstance (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:207:33)
    at resolveParam (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:128:38)
    at async Promise.all (index 0)
    at Injector.resolveConstructorParams (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:143:27)
    at Injector.loadInstance (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:70:13)
    at Injector.loadProvider (api\services\node-app\node_modules\@nestjs\core\injector\injector.js:97:9)
    at api\services\node-app\node_modules\@nestjs\core\injector\instance-loader.js:56:13
    at async Promise.all (index 0)
    at InstanceLoader.createInstancesOfProviders (api\services\node-app\node_modules\@nestjs\core\injector\instance-loader.js:55:9)       
    at api\services\node-app\node_modules\@nestjs\core\injector\instance-loader.js:40:13
    at async Promise.all (index 31)
    at InstanceLoader.createInstances (api\services\node-app\node_modules\@nestjs\core\injector\instance-loader.js:39:9)
    at InstanceLoader.createInstancesOfDependencies (api\services\node-app\node_modules\@nestjs\core\injector\instance-loader.js:22:13)   
    at api\services\node-app\node_modules\@nestjs\core\nest-factory.js:111:17
    at Function.asyncRun (api\services\node-app\node_modules\@nestjs\core\errors\exceptions-zone.js:22:13)
    at NestFactoryStatic.initialize (api\services\node-app\node_modules\@nestjs\core\nest-factory.js:108:13)
    at NestFactoryStatic.create (api\services\node-app\node_modules\@nestjs\core\nest-factory.js:42:9)
    at bootstrap (api\services\node-app\src\main.ts:106:15)
```

**Reply by B4nan** [reply]
So it's clearly coming from your own code, from the `discovery.service.ts`.

**Reply by jdmevo123** [reply]
It’s the @nestjs-plus\discovery service that throws the error, but when I comment out all my modules the app starts successfully, which leads me to believe that there is something in the module that isn’t compatible with mikro orm v6, and not sure how to get a proper error to print to point me in the right direction.

**Reply by B4nan** [reply]
You should report this to them, I cant help you with some 3rd party nest plugins.

**Reply by jdmevo123** [reply]
I have managed to identify this error, and now found what was causing it, this is the stack trace:

There is no duplicate fieldNames, and NewPostCommentNotification.post does not exist in the project at all. I have attached the entity  file below the stack trace.

```typescript
MetadataError: Duplicate fieldNames are not allowed: NewPostCommentNotification.post (fieldName: 'post_id'), NewPostCommentNotification.postId (fieldName: 'post_id')
    at Function.duplicateFieldName (api\services\node-app\node_modules\@mikro-orm\core\errors.js:182:16)
    at MetadataValidator.validateDuplicateFieldNames (api\services\node-app\node_modules\@mikro-orm\core\metadata\MetadataValidator.js:194:42)
    at MetadataValidator.validateEntityDefinition (api\services\node-app\node_modules\@mikro-orm\core\metadata\MetadataValidator.js:39:14)
    at MetadataDiscovery.processEntity (api\services\node-app\node_modules\@mikro-orm\core\metadata\MetadataDiscovery.js:457:24)
    at MetadataDiscovery.processDiscoveredEntities (api\services\node-app\node_modules\@mikro-orm\core\metadata\MetadataDiscovery.js:116:37)
    at MetadataDiscovery.discover (api\services\node-app\node_modules\@mikro-orm\core\metadata\MetadataDiscovery.js:47:14)
    at async MikroORM.discoverEntities (api\services\node-app\node_modules\@mikro-orm\core\MikroORM.js:172:25)
    at async Function.init (api\services\node-app\node_modules\@mikro-orm\core\MikroORM.js:44:9)
    at async Function.handleMigrationCommand (api\services\node-app\node_modules\@mikro-orm\cli\commands\MigrationCommandFactory.js:83:21) {
  entity: undefined
}
``` 

Entity File:
```
import { Cascade, Entity, ManyToOne, Property } from "@mikro-orm/core";
import { InboxNotification } from "./inbox-notification.entity";
import { InboxNotificationTypeEnum } from "../constants/enums";
import { User } from "../../user/entities";
import { PostComment } from "../../post/entities";

@Entity({ discriminatorValue: InboxNotificationTypeEnum.NEW_POST_COMMENT_NOTIFICATION })
export class NewPostCommentNotification extends InboxNotification {
  @Property()
  public readonly postId: number;

  @ManyToOne()
  public readonly to: User;

  @ManyToOne({ cascade: [Cascade.ALL], nullable: true })
  public readonly postComment: PostComment | null = null;

  constructor(to: User, postId: number, postComment?: PostComment) {
    super(InboxNotificationTypeEnum.NEW_POST_COMMENT_NOTIFICATION);

    this.postId = postId;
    this.to = to;
    this.postComment = postComment ?? null;
  }
}

```

**Reply by jdmevo123** [reply]
could this be a problem with packages?

I have tried deleting my migrations folder and recreating, but it throws the same error.

**Reply by B4nan** [reply]
It's a problem with your entity definition.

> There is no duplicate fieldNames,

Yes, there are duplicates, the error explicitly mentions where. You are clearly defining two properties that map to the same column, you can't do that. Mark one of them as `persist: false`.

```diff
-@Property()
+@Property({ persist: false })
public readonly postId: number;
```

Note that you don't need this at all, feels like you are applying typeorm knowledge here. I would suggest you drop the `postId` property completely - you can read the value from `post.id` (and you don't need to populate it for it). I guess the `post` property lives in the parent class, but its surely there - the validation wouldn't kick in without the properties being defined.

**Reply by jdmevo123** [reply]
that worked perfectly thankyou for you help
